opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexerParallelizer.java

```diff
@@ -147,6 +151,15 @@ public ExecutorService getCtagsWatcherExecutor() {
         return result;
     }
 
+    /**
+     * @return the ExecutorService used for repository scan
+     */
+    public ExecutorService getRepositorySearchExecutor() {
+        ExecutorService result = lzRepositorySearchExecutor.get();
+        repositorySearchExecutor = result;
```

Comment:
what is the point of caching what you can get from the lazy supplier?

Is it that you need to keep track if it has ever been initialized?

Reply:
Yes, because you want to avoid initializing something expensive if it hasn't been used elsewhere. I have a branch where I've added an `isActive` property to `LazilyInstantiate` to allow simplifying `IndexerParallelizer`. I can extract that part from the branch and raise a narrow PR.

Reply:
I think it looks cleaner with the PR you just raised. Good job. (#3256)

Reply:
Indeed, rebased.